### PR TITLE
[Rust] Relay bindings

### PR DESCRIPTION
**Pull request by jroesch**
Add full bindings to the Relay AST, including the IRModule.

### CODE REVIEW

**Comment by jroesch** [comment]
cc @mwillsey

**Comment by gussmith23** [comment]
This may not be the correct PR to post this on, but currently I'm having to use TVM Rust bindings as follows:
```sh
$ RUSTFLAGS="-C link-args=-Wl,-rpath,/Users/gus/tvm/build" cargo test
```
I have to explicitly point my @rpath to my TVM location. What's the expected way to do this? Does this seem right?

File: tests/scripts/task_python_docs.sh [hunk 1]
```diff
@@ -68,13 +68,19 @@ npm install
 npm run typedoc
 cd ..
 
+# Rust doc
+cd rust
+cargo doc --workspace --no-deps
```
Comment (binarybana) [inline tests/scripts/task_python_docs.sh:73]: 🚀

**Comment by mwillsey** [comment]
@gussmith23 unless things have changed on this branch, these work for me to build all of TVM and the rust bindings
```
export TVM_HOME= ...your tvm path...
export LLVM_CONFIG_PATH=$(which llvm-config)
export LD_LIBRARY_PATH="$TVM_HOME/build/"
export PYTHONPATH="$TVM_HOME/python:$TVM_HOME/topi/python:${PYTHONPATH}"
```

**Comment by gussmith23** [comment]
Hm, that mostly works, but I can't get rid of these warnings:
```
warning: couldn't execute `llvm-config --prefix` (error: No such file or directory (os error 2))
warning: set the LLVM_CONFIG_PATH environment variable to a valid `llvm-config` executable
```

**Comment by gussmith23** [comment]
What's the thought on how tensor types should be represented? That's something I do need that doesn't seem to be implemented.

**Comment by jroesch** [comment]
@gussmith23 are you using an old version of LLVM?

**Comment by jroesch** [comment]
Closing in favor of #6563

**Pull request closed without merging**